intro-js.md

```diff
@@ -0,0 +1,149 @@
+# Introduction
+Welcome to the weird and wonderful world of JavaScript development. 
+JavaScript, as a language, is used primarily in the development of web applications and websites. 
+
+## Why is JavaScript weird? 
+
+1. Lack of types
+
+JavaScript is a *loosely typed* or *dynamically typed* language
+
+```javascript
+var a = 42; //a is a number here
+var a = 'random' //a is a string here
+var a = true //a is a boolean here
+```
+
+This makes JavaScript development cumbersome and bug ridden, but there are improvements being made. 
+
+A solution to this has been provided through [TypeScript](https://www.typescriptlang.org/docs/home.html)
+
+Angular 2 utilizes TypeScript extensively.
+
+2. Lack of classes
+JavaScript did not have the concept of classes when it was first written.
+
+Classes have now been provided as part of [ECMAScript 2015](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes), but this is just syntactic sugar. 
+
+Example class
+```javascript 
+class Shape {
+    constructor(param1, param2) {
```

Comment:
I'm just thinking in terms of learning javascript from the ground up. It is pretty rare that people use javascript with an object oriented approach when using it for web development. The only time I ever use classes is with React and even with React I try to use functional components whenever I can. I think teaching some es6 concepts are important, but I think it's more important to know vanilla javascript before teaching what is literally pseudo classes. @maneesht what are your thoughts on this?